### PR TITLE
Fix comments / doc-comments of local exceptions in Python

### LINKED ISSUE #2419: Add comment describing who can throw these exceptions
It would be nice to add a comment describing who can throw these exceptions.

The NotExist exceptions and Unknown exceptions: 
 - application code
 - Ice for Python code
 - IcePy C++ extension code

All the other exceptions: only the IcePy C++ extension code?

Maybe users who implement a ValueFactoryManager are expected to throw AlreadyRegisteredException as well from their Python code? 
(seems undesirable - it's better if we can avoid having users throw local exceptions)

_Originally posted by @bernardnormier in https://github.com/zeroc-ice/ice/pull/2417#discussion_r1667030816_

### PR DESCRIPTION
**Pull request by bernardnormier**
Fixes #2419.

### CODE REVIEW

**Review by externl — APPROVED** [review]

File: python/python/Ice/LocalExceptions.py [hunk 1]
```diff
@@ -63,58 +61,47 @@ def operation(self):
...
+    The dispatch could not find the operation carried by the request on the target servant. This is typically due
+    to a mismatch in the Slice definitions, such as the client using Slice definitions newer than the server's.
     """
 
 class UnknownException(LocalException):
     """
-    This exception is raised if an operation call on a server raises an unknown exception. For example, for C++, this
-    exception is raised if the server throws a C++ exception that is not directly or indirectly derived from
-    Ice::LocalException or Ice::UserException.
+    The dispatch failed with an exception that is not a LocalException or a UserException.
```
Comment (pepone) [inline python/python/Ice/LocalExceptions.py:82]:
```suggestion
    The dispatch failed with an exception that is not a LocalException or an UserException.
```

File: python/python/Ice/LocalExceptions.py [hunk 2]
```diff
@@ -63,58 +61,47 @@ def operation(self):
...
 
 @final
 class UnknownUserException(UnknownException):
     """
-    An operation raised an incorrect user exception. This exception is raised if an operation raises a user exception
-    that is not declared in the exception's throws clause. Such undeclared exceptions are not transmitted
-    from the server to the client by the Ice protocol, but instead the client just gets an UnknownUserException.
-    This is necessary in order to not violate the contract established by an operation's signature: Only local
-    exceptions and user exceptions declared in the throws clause can be raised.
+    The dispatch returned a UserException that was not declared in the operation's exception specification.
```
Comment (pepone) [inline python/python/Ice/LocalExceptions.py:94]:
```suggestion
    The dispatch returned an UserException that was not declared in the operation's exception specification.
```

**Review by pepone — APPROVED** [review]

File: python/python/Ice/LocalExceptions.py (hunk 1 above)
Comment (externl) [reply python/python/Ice/LocalExceptions.py:82]: I think `a` is correct here. User doesn't start with a vowel sound.

File: python/python/Ice/LocalExceptions.py (hunk 2 above)
Comment (externl) [reply python/python/Ice/LocalExceptions.py:94]: Ditto.

**Pull request merged**